lib/spack/spack/test/audit.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import pytest
+
+import spack.audit
+import spack.config
+
+
+@pytest.mark.parametrize('packages,failing_check', [
+    # A non existing variant is used in a conflict directive
+    (['wrong-variant-in-conflicts'], 'PKG-DIRECTIVES'),
+    # The package declares a non-existing dependency
+    (['missing-dependency'], 'PKG-DIRECTIVES'),
+    # The package use a non existing variant in a depends_on directive
+    (['wrong-variant-in-depends-on'], 'PKG-DIRECTIVES'),
+    # This package has no issues
+    (['mpileaks'], None)
+])
+def test_package_audits(packages, failing_check, mock_packages):
+    reports = spack.audit.run_group('packages', pkgs=packages)
+
+    for check, errors in reports:
+        # Check that we have errors only if there is an expected failure
+        # and that the tag matches our expectations
+        if bool(failing_check):
+            assert check == failing_check
+            assert errors
+        else:
+            assert not errors
+
+
+@pytest.mark.parametrize('config_section,data,failing_check', [
+    # Double compiler definitions in compilers.yaml
+    ('compilers', [
+        {'compiler': {
```

Comment:
Could we take a non-parameterization approach here? I'm getting lost reading 3-valued tuple spanning 30 lines.

Reply:
I just checked in a text editor and I concur with you. I am used to see these values within an IDE where they look like:
![Screenshot from 2021-06-15 12-39-04](https://user-images.githubusercontent.com/4199709/122039204-b5b27900-cdd6-11eb-9a11-b52411036891.png)

Instead of avoiding parametrization (since the test will be exactly the same) I'll extract the data and bind it to a private variable.

Reply:
Should be addressed by 16f38b6